src/Workspaces/Core/Portable/CodeGeneration/Symbols/CodeGenerationTypeSymbol.cs

```diff
@@ -38,7 +38,7 @@ public virtual ImmutableArray<INamedTypeSymbol> Interfaces
         public ImmutableArray<INamedTypeSymbol> AllInterfaces
             => ImmutableArray.Create<INamedTypeSymbol>();
 
-        public bool IsReferenceType => false;
+        public bool IsReferenceType => TypeKind == TypeKind.Class;
```

Comment:
Which scenario is affected by this change?


Should this method return 'true' if TypeKind is Interface, for example?

Reply:
I'll adjust.
Elsewhere in code generators I made a change that adds a check for `IsReferenceType`. That code relied on a cast to `RecordDeclarationSyntax` which would obviously break.